lib/state/index.js

```diff
@@ -11,11 +11,13 @@ import persistState from 'redux-localstorage';
 import appState from '../flux/app-state';
 
 import auth from './auth/reducer';
+import revisions from './revisions/reducer';
 import settings from './settings/reducer';
 
 export const reducers = combineReducers( {
 	appState: appState.reducer.bind( appState ),
 	auth,
+	revisions,
```

Comment:
> wouldn't it be quite heavy performance-wise?

that depends on how they are stored. if they are full copies of the notes that we asynchronously grab from IndexedDB then yes of course. if however we can grab a list of indexes for valid revisions for the note and they are little more than just integers then probably not.